apps/user-ms/src/main/java/it/pagopa/selfcare/user/service/UserServiceImpl.java

```diff
@@ -303,7 +304,7 @@ public Uni<Void> createOrUpdateUser(CreateUserDto userDto) {
      * Once the userInstitution model is updated or created, attempts to persist the user institution.
      * If any of the operations fail, the method logs an error message and returns a Uni that emits a failure.
      */
-    private Uni<Void> updateUserOnUserRegistryAndUserInstitution(UserResource userResource, CreateUserDto userDto) {
+    private Uni<String> updateUserOnUserRegistryAndUserInstitutionByFiscalCode(UserResource userResource, CreateUserDto userDto) {
         log.info("Updating user on userRegistry and userInstitution");
         String mailUuid = userUtils.getMailUuidFromMail(userResource.getWorkContacts(), userDto.getUser().getInstitutionEmail())
                 .orElse(MAIL_ID_PREFIX + UUID.randomUUID());
```

Comment:
you could define once a logic in a class variable Function and use it all methods....for ex:
Supplier<String> randomMailPrefix = () -> (MAIL_ID_PREFIX + UUID.randomUUID())